cmd/kubelet/app/server.go

```diff
@@ -423,8 +424,12 @@ func Run(ctx context.Context, s *options.KubeletServer, kubeDeps *kubelet.Depend
 	return nil
 }
 
+// checkPermissions confirms that, on linux, the UID of the kubelet process is 0.  For windows, it will return an error automatically.  For other systems, the behaviour is unspecified.
```

Comment:
Can't we skip calling this function at the [call site](https://github.com/kubernetes/kubernetes/blob/199d48728f33692dc5a5e8b808a064bd3e40b51a/cmd/kubelet/app/server.go#L757) and return error there itself if kubelet is running on Windows?

Reply:
We could but I think it's still valid to return an error here, because it's still certainly a meaningless operation to check the UID on windows anyway :)...

I'd say Maybe a follow on PR to revise  the call site ? 

Or do you think I should add that to this PR?